### PR TITLE
feat: Build pagination functions

### PR DESCRIPTION
**Pull request by claudiachua**
Internal JIRA Link: [DI-1401](https://issues.corp.twilio.com/browse/DI-1401)
Related PR: https://github.com/twilio/twilio-oai-generator/pull/250

### CODE REVIEW

File: lib/rest/api/v2010/account/incomingPhoneNumber.js [hunk 1]
```diff
@@ -106,65 +106,7 @@ IncomingPhoneNumberList = function IncomingPhoneNumberList(version, accountSid)
...
-          onComplete();
-        }
-      });
-
-      promise.catch(onComplete);
-    }
-
-    fetchNextPage(_.bind(this.page, this, _.merge(opts, limits)));
-  };
+  IncomingPhoneNumberListInstance.each = version.each;
```
Comment (claudiachua) [inline lib/rest/api/v2010/account/incomingPhoneNumber.js:109]: Had to modify the generated file to test out the new pagination function in oai-generator as a temporary solution (waiting the fix in DI-2334)

File: lib/base/Version.js [hunk 2]
```diff
@@ -183,4 +183,70 @@ Version.prototype.readLimits = function(opts) {
   };
 };
 
+Version.prototype.isCallbackFunction = function(operationPromise, callback) {
+  if (typeof callback === 'function') {
+    operationPromise = operationPromise
+      .then(value => callback(null, value))
+      .catch(error => callback(error));
+  }
```
Comment (childish-sambino) [inline lib/base/Version.js:191]: `is...` functions generally just test something and return a boolean. The name should reflect the actions taken.

File: lib/base/Version.d.ts [hunk 3]
```diff
@@ -69,6 +69,10 @@ declare class Version {
    * @param opts Page limit options passed to the request
    */
   readLimits(opts: Version.PageLimitOptions): Version.PageLimit;
+
+  isCallbackFunction(operationPromise: any, callback: any): Promise<any>;
+  each(params?: any, callback?: any): void;
+  list(params?: any, callback?: any): Promise<any>[];
```
Comment (childish-sambino) [inline lib/base/Version.d.ts:75]: Need a more specific type for these arguments.

File: lib/base/Page.js [hunk 4]
```diff
@@ -186,4 +186,21 @@ Page.prototype.loadPage = function(payload) {
...
 };
 
+function forOwn(obj, iteratee) {
+  obj = Object(obj);
+  for (const [key, val] of Object.entries(obj)) {
+    iteratee(val, key, obj);
+  }
+}
+
+Page.prototype.toJSON = function toJSON() {
```
Comment (childish-sambino) [inline lib/base/Page.js:196]: Why is this needed?

**Review by childish-sambino — APPROVED** [review]

File: lib/base/Page.js (hunk 4 above)
Comment (claudiachua) [reply lib/base/Page.js:196]: toJSON is part of api Page class. Since it was the same across all apis. I had it extracted out like `each`, `list` in Version. https://github.com/twilio/twilio-node/blob/main/lib/rest/api/v2010/account/message.js#L508-L515
